agentos/component.py

```diff
@@ -102,36 +127,69 @@ def from_registry(
 
         return components[identifier]
 
-    @staticmethod
-    def from_registry_file(yaml_file: str, name: str, version: str = None):
+    @classmethod
+    def from_registry_file(
+        cls, yaml_file: str, name: str, version: str = None
+    ) -> "Component":
         registry = Registry.from_yaml(yaml_file)
-        return Component.from_registry(registry, name, version)
+        return cls.from_registry(registry, name, version)
 
-    @staticmethod
+    @classmethod
     def from_class(
+        cls,
         managed_cls: Type[T],
         name: str = None,
         dunder_name: str = None,
+        instantiate: bool = True,
     ) -> "Component":
         name = name if name else managed_cls.__name__
-        return Component(
+        if (
+            managed_cls.__module__ == "__main__"
+        ):  # handle classes defined in REPL.
+            repo = LocalRepo(name)
+            src_file = repo.get_local_repo_dir() / f"{name}.py"
+            with open(src_file, "w") as f:
+                f.write(dill_getsource(managed_cls))
```

Comment:
Is there anyway this write unexpectedly overwrites something not specifically created/controlled by PCS?  Maybe we should check that `src_file` does not exist (if that actually is an invariant)?

Reply:
Good point! Updated to fail if the file exists already.

Also, note that this uses a `LocalRepo`, which takes a `local_dir` argument. The way we use that here (line 149) we end up using the default value for `local_dir` which is the following:
```python
local_dir = "./.pcs_local_repo"
```

...so hopefully using this folder provides enough of a unique namespace that we won't have a lot of failures due to encountering other files the dev has around.

Reply:
BTW, I'm not super happy with this...

See comment here in `repo.py`:

https://github.com/agentos-project/agentos/blob/aecb582389b83f68f394e4ac43f1631975137170/agentos/repo.py#L323-L329